### DISCUSSION TITLE
Load route models with relations via attribute

### DISCUSSION THREAD

**Discussion by kylemilloy** [laravel/framework#53770] [Ideas]
I dunno if this is possible and haven't really dug into it but was just thinking it'd be nice to be able to load relations with route model binding without having to do the extra steps of getting the ID, giving a "with" and then loading it inside the controller body. I find sometimes I have relations that I don't always want to include but then for specific pages I need to bind them.

So like instead of:

```php
public function show(int $user) {
  User::with('orders')->find($user);
  // ...
}
```

Or with two trips...

```php
public function show(User $user) {
  $user->load('orders');
}
```

I can do it in the function args. 

```php
public function show(#[With('orders')] User $user) {
  // ...
}
```

Thoughts?

**Comment by henzeb** [comment]
This might be something you are looking for:

https://laravel.com/docs/11.x/routing#explicit-binding

In the callback you could check for the current route to be the one you want to use it for.

**Reply by kylemilloy** [reply]
Enhhh...kinda but not really? I don't want it hidden in another file and disconnected with the controller behind a bunch of logic...I'd rather it be right in the controller.

**Reply by ghost** [reply]
@kylemilloy In general I see people wanting to hide things under the carpet while using laravel. Why is that? Isn't it easier to understand the code if you see the logic in your file rather that having to dig the logic in laravel files? You can also handle the query errors if you handle it in your files vs letting laravel handle (not really) them for you in their happy flow-ish way.

**Reply by henzeb** [reply]
Well, it's not about "wanting to hide things under the carpet". It is about "I dunno if this is possible and haven't really dug into it". if you haven't dug into it, You get responses that shows you how you can do it in the current state. If you want to have something new, like the proposal of a new attribute, you should make sure you have dug into it and explain why those options are bad. Then we know we don't have to propose the existing options and discuss a possible new feature. And maybe you can implement it and make a PR for it.

**Reply by kylemilloy** [reply]
Jeeez guys lol. I'm just thinking API-first here instead of spending hours looking into it, writing the PR, and then potentially getting it rejected. This post is testing the waters to see if this is a feature people might be interested in other than me. 

If I'm the only one who has ever gone, "Hey wouldn't it be nice to optionally load relations outside of your function body so it can just focus on the biz logic but co-locate that logic in the controller so you can still keep tabs on what it's doing" then there's not much point in writing a PR, right?

**Comment by shaedrich** [comment]
Might be something to add to the [`RouteParameter` attribute](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Container/Attributes/RouteParameter.php) as an additional parameter (might fit into Symfony's [`#Route()` attribute](https://symfony.com/doc/current/routing.html#configuration-block-tabpanel-ecd6010f6252d2743284e9a4ab9a7408bd07a52f) as well) 🤔
